backend/src/integrations/apps.ts

```diff
@@ -760,4 +765,107 @@ const getAppsCodefresh = async ({
   return apps;
 
 };
+
+
+/**
+ * Return list of projects for Windmill integration
+ * @param {Object} obj
+ * @param {String} obj.accessToken - access token for Windmill API
+ * @returns {Object[]} apps - names of Windmill workspaces
+ * @returns {String} apps.name - name of Windmill workspace
+ */
+const getAppsWindmill = async ({ accessToken }: { accessToken: string }) => {
```

Comment:
Hi @rubenfiszel!

This function is meant to fetch back a list of apps in Windmill that the user can select from to sync their secrets to.

Wondering is this the best way to check that the user under the `accessToken` has write access to user/folder-path based variables? The current implementation attempts to write a sample variable for both cases and keeps apps where the user has sufficient permissions but this seems rather lengthy.

Perhaps there is a better way to check for sufficient permissions rather than having to make a sample secret for each app?

Reply:
The permissions are too dynamic in windmill for this to work well. I would just list the workspaces like you do and assume one has write access on all the secrets he can see.

Then if the write-access is not working, then there should be a clear error during sync.

In most cases, the sync will be done by admins so this won't be a common case.

Reply:
I think it would even be fair to require that the token is one of an admin and thus mention that behavior is somewhat undefined if that's not the case. It's not as precise as it could be but there won't be a lot of cases where a non-admin would sync workspace-level secrets.